cmd/benchmark.go

```diff
@@ -1,267 +1,80 @@
-// 		default:
-// 			panic(fmt.Sprintf("Unknown benchmark strategy: %v", benchmarkStrategy))
-// 		}
-//
-// 		for results.Measured < expectedMeasured {
-// 			select {
-// 			case s := <-sentCh:
-// 				log.Infof("[benchmark] request chainID=%s", s.ObjectId)
-// 				results.Total = results.Total + 1
-// 			case r := <-resultsCh:
-// 				log.Infof("[benchmark] results chainID=%s duration=%dms err=%s", r.Transaction.ObjectId, r.Duration, r.Error)
-// 				results.Measured = results.Measured + 1
-//
-// 				if r.Error != "" {
-// 					results.Errors = append(results.Errors, r.Error)
-// 					results.Failures = results.Failures + 1
-// 				} else {
-// 					results.Durations = append(results.Durations, r.Duration)
-// 					results.Successes = results.Successes + 1
-// 				}
-// 			}
-// 		}
-//
-// 		sum := 0
-// 		for _, v := range results.Durations {
-// 			sum = sum + v
-// 		}
-//
-// 		if results.Successes > 0 {
-// 			results.AverageDuration = sum / results.Successes
-//
-// 			sorted := make([]int, len(results.Durations))
-// 			copy(sorted, results.Durations)
-// 			sort.Ints(sorted)
-//
-// 			results.MinDuration = sorted[0]
-// 			results.MaxDuration = sorted[len(sorted)-1]
-// 			p95Index := int64(math.Round(float64(len(sorted))*0.95)) - 1
-// 			results.P95Duration = sorted[p95Index]
-// 		}
-//
-// 		out, _ := json.MarshalIndent(results, "", "  ")
-// 		fmt.Println(string(out))
-// 	},
-// }
-//
-// func init() {
-// 	rootCmd.AddCommand(benchmark)
-// 	benchmark.Flags().IntVarP(&benchmarkConcurrency, "concurrency", "c", 1, "how many transactions to execute at once")
-// 	benchmark.Flags().IntVarP(&benchmarkDuration, "iterations", "i", 10, "how many seconds to run benchmark for")
-// 	benchmark.Flags().IntVarP(&benchmarkTimeout, "timeout", "t", 0, "seconds per transaction to wait before timing out")
-// 	benchmark.Flags().IntVarP(&benchmarkStartDelay, "start-delay", "d", 0, "how many seconds to wait before kicking off the benchmark; useful if network needs to stablize first")
-// 	benchmark.Flags().StringVarP(&benchmarkStrategy, "strategy", "s", "", "(tps|tps-no-ack) - tps will measure every transaction, tps-no-ack will only measure the last round of transactions")
-// 	benchmark.Flags().BoolVar(&enableJaegerTracing, "jaeger-tracing", false, "enable jaeger tracing")
-// 	benchmark.Flags().BoolVar(&enableElasticTracing, "elastic-tracing", false, "enable elastic tracing")
-// 	benchmark.Flags().StringVar(&tracingTagName, "tracing-name", "", "adds tag to tracing for lookup")
-// }
+	"github.com/quorumcontrol/tupelo-go-sdk/gossip/client"
+	"github.com/quorumcontrol/tupelo-go-sdk/gossip/client/pubsubinterfaces/pubsubwrapper"
+	"github.com/quorumcontrol/tupelo-go-sdk/p2p"
+	"github.com/quorumcontrol/tupelo/gossip/benchmark"
+	"github.com/quorumcontrol/tupelo/nodebuilder"
+	"github.com/spf13/cobra"
+)
+
+var benchmarkConcurrency int
+var benchmarkTimeout int
+var benchmarkDuration int
+var benchmarkStartDelay int
```

Comment:
I removed the redundant `benchmark` prefix from all of these in my start on this. Take or leave it. :)

Reply:
well - the vars aren't local to this file, they are local to the "cmd" package and so if any other packages had something like "timeout" then that could be confusing.

Reply:
Ah, yeah, good point. 👍 to leaving the prefix then.